### PR TITLE
Feature/workflows

### PR DESCRIPTION
**Pull request by spau6**
Added the CD workflow to include deploy and e2e runs

### CODE REVIEW

File: .github/workflows/c8y-deploy-action/action.yml
Comment (ghost) [inline .github/workflows/c8y-deploy-action/action.yml:1]: First create a folder called `actions` in .github folder and add the actions there. Even after this change if you have to modify execute access to the script, create a separate action in a different repository and use it.

File: .github/workflows/widget-cd.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          version=`jq -r '.version' package.json`
+          echo "app_file_name=${{env.APP_CONTEXT_PATH}}-$version" >> $GITHUB_OUTPUT
+
+      - name: Download webapp
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ steps.compose-file-name.outputs.app_file_name }}
+          path: "./downloads"
+
+      - name: Grant permission
```
Comment (ghost) [inline .github/workflows/widget-cd.yml]: Refer previous comment to remove this step

**Review by ghost — DISMISSED** [review]

File: .github/workflows/c8y-deploy-action/action.yml
Comment (spau6) [reply .github/workflows/c8y-deploy-action/action.yml:1]: added a folder called action and added the action.yml in it. Action is added with permission step

File: .github/workflows/widget-cd.yml (hunk 1 above)
Comment (spau6) [reply .github/workflows/widget-cd.yml]: done

**Review by amar-c8y — APPROVED** [review]

**Review by dchc01 — APPROVED** [review]

**Pull request merged**